keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -88,6 +90,51 @@ class SqlDeliveryConfigRepository(
     return deliveryConfigUIDs.size
   }
 
+  override fun delete(name: String) {
```

Comment:
should this be in a `jooq.transaction`?

Reply:
I think so. It's hard enough that we have to write all this jooq/sql code to cascade deletes and retrievals of object trees (which I feel is pretty prone to implementation mistakes), so I think we definitely need to wrap things in transactions when operating on multiple tables.

Reply:
Its very important that we add retry logic for every sql write as we implement multi-table transactions like this. They will result in deadlocks under the right sort of concurrency - that is normal and ok, but keel can't currently handle them at all.

Reply:
A PR for sql retries: https://github.com/spinnaker/keel/pull/741